tests/unit/models/gpflow/test_models.py

```diff
@@ -323,7 +328,7 @@ def test_gpflow_models_cached_predictions_correct(
     model, _ = gpflow_interface_factory(x, y)
 
     if after_model_optimize:
-        model._optimizer = BatchOptimizer(tf.optimizers.SGD(), max_iter=1)
+        model._optimizer = BatchOptimizer(gpflow.optimizers.Scipy(), max_iter=1)
```

Comment:
`BatchOptimizer` is meant to be used for stochastic optimizers in TF, why this change? was this for the VGP model that we have discussed?

Reply:
Yes, it's due to https://github.com/secondmind-labs/trieste/issues/754. We can chat about how important we think working around this is in the next planning meeting.